src/pymodule.cpp

```diff
@@ -12,6 +13,66 @@ using tesseract::PageSegMode;
 using tesseract::ResultIterator;
 using tesseract::TessBaseAPI;
 
+static void CopyRowMajor(uint8_t* dst, const py::buffer_info& info) {
```

Comment:
Hi, sorry that I could not understand how this code works. Could you do it without optimisation? Copying data in row major format should not be that complicated IMO but I am happy to be corrected.

Reply:
this is not copying data *in* row major format, it is copying data in an arbitrary memory layout (given by the shape and strides) *into* row major format, which is what leptonica needs. I agree that my code isn't super readable, but I don't think i can be simplified much more (unless you want to specialize it to 2 or 3 dimensions, which is an option).

Reply:
To throw out a hypothetical, are we anticipating 20 dimensional images going through this thing? If no, then specialisation doesn't seem like such a bad idea.

Reply:
granted, but specializing to 3 dimensions means we lose the important coercion optimization (i.e. do everything with a single memcpy() if the layout allows for it). to maintain coercion optimization, we'd need to write versions for 0, 1, 2 and 3 dimensions